src/main/java/at/hannibal2/skyhanni/features/misc/TabWidgetSettings.kt

```diff
@@ -0,0 +1,109 @@
+package at.hannibal2.skyhanni.features.misc
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.events.GuiContainerEvent
+import at.hannibal2.skyhanni.events.InventoryCloseEvent
+import at.hannibal2.skyhanni.events.InventoryFullyOpenedEvent
+import at.hannibal2.skyhanni.utils.InventoryUtils.getInventoryName
+import at.hannibal2.skyhanni.utils.InventoryUtils.getUpperItems
+import at.hannibal2.skyhanni.utils.ItemUtils.getLore
+import at.hannibal2.skyhanni.utils.LorenzColor
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.NEUInternalName
+import at.hannibal2.skyhanni.utils.RenderUtils.highlight
+import at.hannibal2.skyhanni.utils.StringUtils.anyMatches
+import at.hannibal2.skyhanni.utils.StringUtils.matches
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraft.client.gui.inventory.GuiChest
+import net.minecraft.inventory.ContainerChest
+import net.minecraft.inventory.Slot
+import net.minecraft.item.Item
+import net.minecraft.item.ItemStack
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+class TabWidgetSettings {
+    private val patternGroup = RepoPattern.group("tab.widget.setting")
+    private val mainPageSettingPattern by patternGroup.pattern(
+        "gui",
+        "WIDGETS.*"
+    )
+    private val mainPageWidgetPattern by patternGroup.pattern(
+        "main",
+        "§7Currently:.*"
+    )
+    private val subPageWidgetPattern by patternGroup.pattern(
+        "sub",
+        "§eClick to .*"
+    )
+    private val shownSettingPattern by patternGroup.pattern(
+        "show",
+        "Shown .* Setting.* |.*Widget Settings"
+    )
+    private val clickToDisablePattern by patternGroup.pattern(
+        "click.disable",
+        ".*(disable!)"
+    )
+    private val enabledPattern by patternGroup.pattern(
+        "is.enabled",
+        ".*(ENABLED)"
+    )
+
+    var inInventory = false;
```

Comment:
This variable is not used. Either remove it or use it in `GuiContainerEvent.BackgroundDrawnEvent`
Since the map `highlights` gets cleared on inventory close, this extra boolean not working is not game breaking. But having this variable is a nice consistency bonus and reduces `GuiContainerEvent.BackgroundDrawnEvent` performance drain while not in use.